codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -265,18 +271,21 @@ public Http2Stream resetSent() {
         }
 
         @Override
-        public final Object setProperty(Object key, Object value) {
-            return data.put(key, value);
+        public final <K, V> V setProperty(PropertyKey<K, V> key, V value) {
+            checkNotNull(key, "key");
+            return properties.add((DefaultPropertyKey) key, value);
```

Comment:
The previous discussion was collapsed...so I'll start again...

I don't think we want to be casting to `DefaultPropertyKey` here.  The `PropertyKey` is publicly visible and the user could supply any implementation they want.  If there are methods we are depending upon for our algorithms perhaps we should move these methods into the `PropertyKey` interface.